rotkehlchen/db/eth2.py

```diff
@@ -19,12 +19,37 @@
 logger = logging.getLogger(__name__)
 log = RotkehlchenLogsAdapter(logger)
 
+WITHDRAWALS_RECHECK_PERIOD = 3 * 3600 * 1000  # 3 hours in milliseconds
+
 
 class DBEth2():
 
     def __init__(self, database: 'DBHandler') -> None:
         self.db = database
 
+    def get_validators_to_query_for_withdrawals(self, up_to_tsms: TimestampMS) -> list[tuple[int, Timestamp]]:  # noqa: E501
+        """Gets a list of validators that need to be queried for new withdrawals
+
+        Validators need to be queried if last time they are queried was more than X seconds.
+
+        Returns a list of tuples. First entry is validator index and second entry is
+        last queried timestamp for daily stats of that validator.
+        """
+        query_str = """
+            SELECT V.validator_index, E.timestamp FROM eth2_validators V
+            LEFT JOIN eth_staking_events_info S ON S.validator_index = V.validator_index
+            LEFT JOIN history_events E ON
+            E.identifier = S.identifier WHERE ? - (SELECT MAX(timestamp) FROM history_events WHERE identifier=S.identifier) > ? AND S.validator_index IS NOT NULL
+            UNION
+            SELECT DISTINCT V2.validator_index, 0 FROM eth2_validators V2 WHERE
+            V2.validator_index NOT IN (SELECT validator_index FROM eth_staking_events_info)
+        """  # noqa: E501
+        cursor = self.db.conn.cursor()
+        result = cursor.execute(
+            query_str, (up_to_tsms, WITHDRAWALS_RECHECK_PERIOD),
+        )
```

Comment:
This fits in one line

Reply:
Heh ... this is what you see? Check the line above.